### PR TITLE
Remove gapi preproc

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Removed GAPI preprocessing library from OV
 - Removed I420 and NV12 blobs
 - Removed deprecated legacy API

### Tickets:
 - CVS-91368

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,6 @@ add_subdirectory(plugins)
 add_subdirectory(inference)
 include(cmake/openvino.cmake)
 # preprocessing has dependency on `openvino` for static build
```
Comment (ilya-lavrenov) [inline src/CMakeLists.txt:27]: remove comment as well

File: src/inference/dev_api/ie_preprocess_data.hpp [hunk 2]
```diff
@@ -4,19 +4,19 @@
```
Comment (ilya-lavrenov) [inline src/inference/dev_api/ie_preprocess_data.hpp:4]: should we remove this file as well?

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
We also have tests on dynamic preprocessing - they should be removed as well

File: src/inference/include/ie/ie_compound_blob.h [hunk 3]
```diff
@@ -115,162 +115,6 @@ class INFERENCE_ENGINE_API_CLASS(CompoundBlob) : public Blob {
     const std::shared_ptr<IAllocator>& getAllocator() const noexcept override;
 };
 
-/**
- * @brief Represents a blob that contains two planes (Y and UV) in NV12 color format
- */
-class INFERENCE_ENGINE_DEPRECATED("This class is deprecated and will be removed in 2023.1 release")
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply src/inference/include/ie/ie_compound_blob.h:121]: This file contains batched blob which is still used and not deprecated.

**Comment by ilyachur** [comment]
> We also have tests on dynamic preprocessing - they should be removed as well

I am waiting for CI in order to face all places which we need to fix

File: src/inference/include/ie/ie_compound_blob.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/inference/include/ie/ie_compound_blob.h:121]: but batched blob was required to pass several NV12 blobs into SetBlob.
So, no NV12 / I420 => no needs to have BatchedBlob (as compound blob as well).

File: src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp [hunk 4]
```diff
@@ -284,54 +260,7 @@ void InferRequestLegacy::SetBlob(const std::string& name, const Blob::Ptr& data)
...
-                            _deviceInputs[uv_name] = nv12_ptr->uv();
-                            is_remote = true;
-                        }
-                    }
-                }
-            }
-            if (is_remote)
-                _inputs[name] = data;
         }
-
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp:334]: Shouldn't `preProcessingRequired ` below be removed as well?

File: src/inference/include/ie/gpu/gpu_config.hpp [hunk 5]
```diff
@@ -99,12 +99,6 @@ DECLARE_GPU_CONFIG_VALUE(HOST_TASK_PRIORITY_HIGH);
 DECLARE_GPU_CONFIG_VALUE(HOST_TASK_PRIORITY_MEDIUM);
 DECLARE_GPU_CONFIG_VALUE(HOST_TASK_PRIORITY_LOW);
 
-/**
- * @brief This key should be set to correctly handle NV12 input without pre-processing.
- * Turned off by default.
- */
-DECLARE_GPU_CONFIG_KEY(NV12_TWO_INPUTS);
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/gpu/gpu_config.hpp:106]: should we mark it as deprecated in 2023.0 ?

File: samples/python/hello_query_device/README.md [hunk 6]
```diff
@@ -62,7 +62,6 @@ For example:
    [ INFO ]                CPU_THREADS_NUM: 0
    [ INFO ]                CPU_THROUGHPUT_STREAMS: 1
    [ INFO ]                DUMP_EXEC_GRAPH_AS_DOT:
-   [ INFO ]                DYN_BATCH_ENABLED: NO
    [ INFO ]                DYN_BATCH_LIMIT: 0
```
Comment (ilya-lavrenov) [inline samples/python/hello_query_device/README.md:65]: `DYN_BATCH_LIMIT` should also be removed

**Pull request closed without merging**